### ISSUE TITLE
Getting Message: Invalid credentials (bad signature)

### ISSUE THREAD

**Issue by kilimchoi** [ccxt/ccxt#3519]
ATTENTION!!!

MUST READ THIS BEFORE SUBMITTING ISSUES:

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

- OS: Mac OSX
- Programming Language version:
- CCXT version: 1.17.47
- Exchange: Independent Reserve
- Method: privatePostGetTrades
----------------------------------------------------------------------------------------------------
Hi, 
This is more of a question than raising an issue with the library. But I was wondering how I could get around the problem of getting a message "Invalid credentials (bad signature)" when I'm making a call to the private method for Independent Reserve. 

My code is as follows: 

```
ir = ccxt.independentreserve({
   "apiKey": MY_API_KEY,
   "secret": SECRET
})
ir.privatePostGetTrades({"PageIndex": "1", "PageSize": "5"})

**Comment by kroitor** [comment]
Hi! Unfortunately, I'm having a difficulty with obtaining a key from Independent Reserve, as they won't accept signups from my country (they're too restrictive). If you can share an API key for it, I'll take a look and will get back to you soon with a fix. We don't need your funds or passwords or personal info, just need a temporary key for debugging purposes. A fresh new keypair with zero balance will do just fine. We would be thankful if you could help us get a testing keypair. Thx anyways.

**Comment by kilimchoi** [comment]
You need an api key only? No api secret?

**Comment by kroitor** [comment]
@kilimchoi nope, we need the key pair, I mean, both the key and the secret. We won't need it after debugging, so you could safely delete it after we fix the bug here. Thx again.

**Comment by kilimchoi** [comment]
@kroitor Just sent it

**Comment by kroitor** [comment]
Thx, got it! I'll take a look into it shortly and will get back to you asap with a solution to the problem. Thanks so much for your assistance!

**Comment by kroitor** [comment]
Ok, looks like I've fixed it in version 1.17.49 (it will arrive in 5-10 minutes).

However, because IndependentReserve wants special parameter ordering for each particular private method, the code snippet that you posted above won't work well in Python with `dict`, because Python dicts are **unordered**.

So, instead of the above snippet, I'd recommend to use the following unified method:

```Python
exchange = ccxt.independentreserve({
    'apiKey': 'YOUR_API_KEY',
    'secret': 'YOUR_SECRET',
})
exchange.fetchMyTrades(symbol=None, since=None, limit=5)
```

You can safely delete that key now. Once again, thx for your help! Without it it would be really hard to debug this.

Don't forget to update CCXT. Let me know if you have any difficulties with it, will be happy to help!
